test/e2e/usersignup_test.go

```diff
@@ -0,0 +1,546 @@
+	// 1) the Approved condition is set to true
+	// 2) the Approved reason is set to ApprovedAutomatically
+	// 3) the Complete condition is (eventually) set to true
+	err = s.hostAwait.waitForUserSignupStatusConditions(userSignup.Name,
+		v1alpha1.Condition{
+			Type: v1alpha1.UserSignupApproved,
+			Status: corev1.ConditionTrue,
+			Reason: "ApprovedAutomatically",
+		},
+		v1alpha1.Condition{
+			Type:   v1alpha1.UserSignupComplete,
+			Status: corev1.ConditionTrue,
+		})
+	require.NoError(s.T(), err)
+
+	// Lookup the MUR
+	mur := &v1alpha1.MasterUserRecord{}
+	err = s.awaitility.Client.Get(context.TODO(), types.NamespacedName{Namespace: s.namespace, Name: userSignup.Name}, mur)
+	require.NoError(s.T(), err)
+
+	require.Len(s.T(), mur.Spec.UserAccounts, 1)
+
+	// Confirm the target cluster was set
+	require.NotEmpty(s.T(), mur.Spec.UserAccounts[0].TargetCluster)
+}
+
+func (s *userSignupIntegrationTest) TestDeletedUserSignupIsGarbageCollected() {
+	// Set the user approval policy to automatic
+	s.setApprovalPolicyConfig(config.UserApprovalPolicyAutomatic)
+
+	// Create user signup - no approval set
+	s.T().Logf("Creating UserSignup with namespace %s", s.namespace)
+	userSignup := s.newUserSignup("oliver")
+	err := s.awaitility.Client.Create(context.TODO(), userSignup, &framework.CleanupOptions{TestContext: s.testCtx,
+		Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	require.NoError(s.T(), err)
+	s.T().Logf("user signup '%s' created", userSignup.Name)
+
+	// Confirm the UserSignup was created
+	err = s.hostAwait.waitForUserSignup(userSignup.Name)
+	require.NoError(s.T(), err)
+
+	// Confirm the MasterUserRecord was created
+	err = s.hostAwait.waitForMasterUserRecord(userSignup.Name)
+	require.NoError(s.T(), err)
+
+	// Delete the UserSignup
+	err = s.awaitility.Client.Delete(context.TODO(), userSignup)
+	require.NoError(s.T(), err)
+
+	// Confirm the UserSignup was deleted
+	err = s.hostAwait.waitForUserSignup(userSignup.Name)
+	require.Error(s.T(), err)
+
+	// Confirm the MasterUserRecord was deleted
+	err = s.hostAwait.waitForMasterUserRecord(userSignup.Name)
+	require.Error(s.T(), err)
+}
+
+func (s *userSignupIntegrationTest) TestUserSignupWithAutoApproval() {
+	// Set the user approval policy to automatic
+	s.setApprovalPolicyConfig(config.UserApprovalPolicyAutomatic)
+
+	// Create user signup - no approval set
+	s.T().Logf("Creating UserSignup with namespace %s", s.namespace)
+	userSignup := s.newUserSignup("charles")
+	err := s.awaitility.Client.Create(context.TODO(), userSignup, &framework.CleanupOptions{TestContext: s.testCtx,
+		Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	require.NoError(s.T(), err)
+	s.T().Logf("user signup '%s' created", userSignup.Name)
+
+	// Confirm the UserSignup was created
+	err = s.hostAwait.waitForUserSignup(userSignup.Name)
+	require.NoError(s.T(), err)
+
+	// Confirm the MasterUserRecord was created
+	err = s.hostAwait.waitForMasterUserRecord(userSignup.Name)
```

Comment:
At least in one test we should check the content of MUR too:
- UserID
- UserAccounts
  - userID
  - NSLimit
  - Template
  - ...

Reply:
Added a test for this.